corehq/apps/analytics/tasks.py

```diff
@@ -0,0 +1,65 @@
+from celery.task import task
+import json
+import requests
+import urllib
+
+from settings import ANALYTICS_IDS
+
+
+def _track_on_hubspot(webuser, properties):
+    """
+    Update or create a new "contact" on hubspot. Record that the user has
+    created an account on HQ.
+
+    properties is a dictionary mapping property names to values.
+    Note that property names must exist on hubspot prior to use.
+    """
+    # Note: Hubspot recommends OAuth instead of api key
+    # TODO: Use batch requests / be mindful of rate limit
+
+    api_key = ANALYTICS_IDS.get('HUBSPOT_API_KEY', None)
+    if api_key and not webuser.is_dimagi:
+        req = requests.post(
+            u"https://api.hubapi.com/contacts/v1/contact/createOrUpdate/email/{}".format(
+                urllib.quote(webuser.username)
+            ),
+            params={'hapikey': api_key},
+            data=json.dumps(
+                {'properties': [
+                    {'property': k, 'value': v} for k, v in properties.items()
+                ]}
+            ),
+        )
+        req.raise_for_status()
+
+
+def _get_user_hubspot_id(webuser):
+    api_key = ANALYTICS_IDS.get('HUBSPOT_API_KEY', None)
+    if api_key:
+        req = requests.get(
+            u"https://api.hubapi.com/contacts/v1/contact/email/{}/profile".format(
+                urllib.quote(webuser.username)
+            ),
+            params={'hapikey': api_key},
+        )
+        if req.status_code == 404:
+            return None
+        req.raise_for_status()
+        return req.json().get("vid", None)
+    return None
+
+
+@task(queue='background_queue', acks_late=True)
+def track_created_hq_account_on_hubspot(webuser):
+    _track_on_hubspot(webuser, {
+        'created_account_in_hq': True,
+        'commcare_user': True,
+    })
+
+
+@task(queue='background_queue', acks_late=True)
```

Comment:
I think you can also add `ignore_result` to these since you're not returning anything and don't need to track progress.